### PR TITLE
create tabLayout and viewpager

### PR DESCRIPTION
**Pull request by VuManhHung97**
Create song genres with tabLayut and ViewPager
Handle swipe events to select categories

https://edu-redmine.sun-asterisk.vn/issues/12506

songgenre

### CODE REVIEW

File: app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java [hunk 1]
```diff
@@ -3,18 +3,47 @@
...
+        initView(view);
         return view;
     }
+
+    private void initView(View view) {
+        mTabLayout = view.findViewById(R.id.tabLayoutHome);
+        mViewPager = view.findViewById(R.id.viewPagerHome);
+        FragmentManager manager = getFragmentManager();
+        PagerAdapter pagerAdapter = new PagerAdapter(manager);
+        pagerAdapter.addFragment(new AllSongFragment(), "ALLSongs");
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java]: Hard code.

File: app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java [hunk 2]
```diff
@@ -3,18 +3,47 @@
...
+        pagerAdapter.addFragment(new AudioFragment(), "Audio");
+        pagerAdapter.addFragment(new AmbientFragment(), "Ambient");
+        pagerAdapter.addFragment(new RockFragment(), "Rock");
+        pagerAdapter.addFragment(new ClassicalFragment(), "Classical");
+        pagerAdapter.addFragment(new CountryFragment(), "Country");
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java]: as above

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -4,18 +4,31 @@
...
         app:layout_constraintBottom_toTopOf="@id/navigationMenuHome"
-        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/toolbarHome"
         />
     <android.support.design.widget.BottomNavigationView
         android:id="@+id/navigationMenuHome"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_while"
+        android:background="@color/colorWhile"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
        android:background="@color/colorWhite"
```

File: app/src/main/res/layout/fragment_rock.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    >
+
+</android.support.constraint.ConstraintLayout>
```
Comment (thaohq-0814) [inline app/src/main/res/layout/fragment_rock.xml]: add new line below this

File: app/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <color name="colorPrimary">#008577</color>
+    <color name="colorBlack">#000000</color>
     <color name="colorPrimaryDark">#00574B</color>
     <color name="colorAccent">#D81B60</color>
-    <color name="color_while">#FFFFFF</color>
+    <color name="colorWhile">#FFFFFF</color>
```
Comment (thaohq-0814) [inline app/src/main/res/values/colors.xml]:
```suggestion
    <color name="colorWhite">#FFFFFF</color>
```

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -5,4 +5,6 @@
     <string name="title_library">Library</string>
     <string name="title_favorite">Favorite</string>
     <string name="title_setting">Setting</string>
+
```
Comment (thaohq-0814) [inline app/src/main/res/values/strings.xml]: remove blank

**Comment by VuManhHung97** [comment]
ready

**Review by thaohq-0814 — APPROVED** [review]

**Comment by VuManhHung97** [comment]
ready

File: app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java [hunk 7]
```diff
@@ -3,18 +3,47 @@
...
         View view = inflater.inflate(R.layout.fragment_home, container, false);
+        initView(view);
         return view;
     }
+
+    private void initView(View view) {
+        mTabLayout = view.findViewById(R.id.tabLayoutHome);
+        mViewPager = view.findViewById(R.id.viewPagerHome);
+        FragmentManager manager = getFragmentManager();
+        PagerAdapter pagerAdapter = new PagerAdapter(manager);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/music_65/screen/main/HomeFragment.java]:
```suggestion
        PagerAdapter pagerAdapter = new PagerAdapter(getFragmentManager());
```

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -5,4 +5,11 @@
     <string name="title_library">Library</string>
     <string name="title_favorite">Favorite</string>
     <string name="title_setting">Setting</string>
+    <string name="sub_title_slogan">SoundCloud</string>
+    <string name="titleAudio">Audio</string>
+    <string name="titleAmbient">Ambient</string>
+    <string name="titleRock">Rock</string>
+    <string name="titleClassical">Classical</string>
+    <string name="titleCountry">Country</string>
+    <string name="titleAllSongs">ALLSongs</string>
```
Comment (tuanlvt-1117) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="titleAllSongs">All Songs</string>
```

**Comment by VuManhHung97** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**